alws/crud/release.py

```diff
@@ -258,98 +280,158 @@ async def get_pulp_based_response():
-                pkg_info['repositories'] = [
-                    repos_mapping.get(item) for item in release_repositories
-                ]
-            packages.append(pkg_info)
-            added_packages.append(full_name)
-        await prepare_and_execute_async_tasks()
+    for package in pulp_packages:
+        pkg_name = package['name']
+        pkg_version = package['version']
+        pkg_arch = package['arch']
+        full_name = package['full_name']
+        if full_name in added_packages:
+            continue
+        if pkg_arch == 'noarch':
+            prepare_data_for_executing_async_tasks(package, full_name)
+        key = (pkg_name, pkg_version, pkg_arch)
+        predicted_package = beholder_cache.get(key, [])
+        pkg_info = {'package': package, 'repositories': []}
+        release_repositories = set()
+        repositories = []
+        if not predicted_package:
+            continue
+        repositories = predicted_package['repositories']
+        for repo in repositories:
+            ref_repo_name = repo['name']
+            repo_name = (repo_name_regex.search(ref_repo_name)
+                            .groupdict()['name'])
+            release_repo_name = '-'.join([
+                clean_ref_dist_name_lower,
+                base_platform.distr_version,
+                repo_name
+            ])
+            debug = ref_repo_name.endswith('debuginfo')
+            if repo['arch'] == 'src':
+                debug = False
+            release_repo = RepoType(
+                release_repo_name, repo['arch'], debug)
+            release_repositories.add(release_repo)
+        pkg_info['repositories'] = [
+            repos_mapping.get(item) for item in release_repositories
+        ]
+        added_packages.add(full_name)
+        packages.append(pkg_info)
+
+    for package in pulp_packages:
+        if package['full_name'] in added_packages:
+            continue
+        added_packages.add(package['full_name'])
+        release_repo = repos_mapping[RepoType(
+            '-'.join([
+                clean_ref_dist_name_lower,
+                base_platform.distr_version,
+                'devel'
+            ]),
+            package['task_arch'],
+            False
+        )]
+        pkg_info = {
+            'package': package, 
+            'repositories': [release_repo]
+        }
+        packages.append(pkg_info)
+
+    # if noarch package already in repo with same NEVRA,
+    # we should exclude this repo when generate release plan
+    await prepare_and_execute_async_tasks()
+    for pkg_info in packages:
+        package = pkg_info['package']
+        if package['arch'] != 'noarch':
+            continue
+        repos_ids = existing_packages.get(package['full_name'], [])
+        # TODO: also add here check for build arches
+        new_repos = [
+            repo for repo in pkg_info['repositories']
+            if repo['id'] not in repos_ids
+        ]
+        pkg_info['repositories'] = new_repos
```

Comment:
it's old logic from my previous changes for noarch checking, pls compare ur branch with actual `master` branch and delete old logic